Added EMR on EKS Custom Image CLI

zhuyulin27 opened the pull request:
*Issue #, if available:*

*Description of changes:*

Added brew formula and config file for [amazon-emr-on-eks-custom-image-cli](https://github.com/awslabs/amazon-emr-on-eks-custom-image-cli), also updated ReadMe with custom image cli repo details. EMR on EKS custom image cli enables customers to interact with EMR on EKS custom images via a CLI tool.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

bwagner5 commented on Formula/amazon-emr-on-eks-custom-image-cli.rb:
```diff
@@ -0,0 +1,32 @@
+# -*- coding: utf-8 -*-
+require_relative '../ConfigProvider/config_provider'
+
+class AmazonEmrOnEksCustomImageCli < Formula
```
This depends on python and docker, correct?  I'm not sure about adding docker as a dependency since I'm not sure how many people would be installed docker with homebrew, but what about python?

bwagner5 replied:
Maybe a `caveat` is the best option forward: https://github.com/aws/homebrew-tap/blob/f89532a77230a2914d7e6e0d65041cfd76ad46da/Formula/eks-anywhere.rb#L18-L25 

You can just explain that docker is a dependency

zhuyulin27 replied:
That sounds good! I updated the formula with caveats function, you could review and check that. Thanks!

bwagner5 reviewed (changes requested):
I've tested the install and works fine on my mac.  

Can you change the formula to `EmrOnEksCustomImage` ? The CLI name is already long and it's a bit confusing what the user should execute as the binary name after installing.

zhuyulin27 commented:
> I've tested the install and works fine on my mac.
> 
> Can you change the formula to `EmrOnEksCustomImage` and `emr-on-eks-custom-image` ? The CLI name is already long and it's a bit confusing what the user should execute as the binary name after installing.

Thanks for your suggestion! I've changed the name of the formula to be `emr-on-eks-custom-image.rb' and the class name to be `EmrOnEksCustomImage` and tested locally. (Also updated the path in README)

bwagner5 commented on README.md:
```diff
@@ -25,6 +25,7 @@ brew install <FORMULA>
 | [xray-daemon](https://github.com/aws/aws-xray-daemon) | [formula](Formula/xray-daemon.rb) | Relays traffic to AWS X-Ray API |
 | [aws-simple-ec2-cli](https://github.com/awslabs/aws-simple-ec2-cli) | [formula](Formula/aws-simple-ec2-cli.rb) | CLI tool that simplifies the process of launching, connecting and terminating EC2 instances |
 | [qldbshell](https://github.com/awslabs/amazon-qldb-shell) | [formula](Formula/qldbshell.rb) | A CLI Shell for interacting with and processing queries against AWS QLDB |
+| [amazon-emr-on-eks-custom-image-cli](https://github.com/awslabs/amazon-emr-on-eks-custom-image-cli) | [formula](Formula/emr-on-eks-custom-image.rb) | A CLI tool to interact with EMR on EKS custom images.
```
Make sure to change the link text `[amazon-emr-on-eks-custom-image-cli](https://...)` to `[emr-on-eks-custom-image](https://..)`  so that a user can simply `brew install [name]` and get the name from the readme table.

zhuyulin27 replied:
Oh, I though it should be the repository name so I didn't change it, let me do that.

zhuyulin27 replied:
Done!

bwagner5 reviewed (changes requested):
Just the one README comment. Everything else looks good!

bwagner5 reviewed (approved):
nice! LGTM!
